controllers/csi/driver/server.go

```diff
@@ -0,0 +1,327 @@
+	if err != nil {
+		return fmt.Errorf("failed to parse endpoint '%s': %w", svr.opts.Endpoint, err)
+	}
+
+	if proto == "unix" {
+		if err := os.Remove(addr); err != nil && !os.IsNotExist(err) {
+			return fmt.Errorf("failed to remove old endpoint on '%s': %w", addr, err)
+		}
+	}
+
+	svr.log.Info("Starting listener", "protocol", proto, "address", addr)
+
+	listener, err := net.Listen(proto, addr)
+	if err != nil {
+		return fmt.Errorf("failed to start server: %w", err)
+	}
+
+	server := grpc.NewServer(grpc.UnaryInterceptor(logGRPC(log)))
+	go func() {
+		<-ctx.Done()
+		svr.log.Info("Stopping server")
+		server.GracefulStop()
+		svr.log.Info("Stopped server")
+	}()
+
+	csi.RegisterIdentityServer(server, svr)
+	csi.RegisterNodeServer(server, svr)
+
+	svr.log.Info("Listening for connections on address", "address", listener.Addr())
+
+	server.Serve(listener)
+
+	return nil
+}
+
+// csi.IdentityServer implementation
+
+func (svr *CSIDriverServer) GetPluginInfo(ctx context.Context, req *csi.GetPluginInfoRequest) (*csi.GetPluginInfoResponse, error) {
+	return &csi.GetPluginInfoResponse{Name: dtcsi.DriverName, VendorVersion: version.Version}, nil
+}
+
+func (svr *CSIDriverServer) Probe(ctx context.Context, req *csi.ProbeRequest) (*csi.ProbeResponse, error) {
+	return &csi.ProbeResponse{}, nil
+}
+
+func (svr *CSIDriverServer) GetPluginCapabilities(ctx context.Context, req *csi.GetPluginCapabilitiesRequest) (*csi.GetPluginCapabilitiesResponse, error) {
+	return &csi.GetPluginCapabilitiesResponse{}, nil
+}
+
+// csi.NodeServer implementation
+
+func (svr *CSIDriverServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
+	if req.GetVolumeCapability() == nil {
+		return nil, status.Error(codes.InvalidArgument, "Volume capability missing in request")
+	}
+
+	volID := req.GetVolumeId()
+	if volID == "" {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID missing in request")
+	}
+
+	targetPath := req.GetTargetPath()
+	if targetPath == "" {
+		return nil, status.Error(codes.InvalidArgument, "Target path missing in request")
+	}
+
+	if req.GetVolumeCapability().GetBlock() != nil {
+		return nil, status.Error(codes.InvalidArgument, "cannot have block access type")
+	}
+
+	if req.GetVolumeCapability().GetMount() == nil {
+		return nil, status.Error(codes.InvalidArgument, "expecting to have mount access type")
+	}
+
+	volCtx := req.GetVolumeContext()
+	if volCtx == nil {
+		return nil, status.Error(codes.InvalidArgument, "Publish context missing in request")
+	}
+
+	nsName := volCtx[podNamespaceContextKey]
+	if nsName == "" {
+		return nil, status.Error(codes.InvalidArgument, "No namespace included with request")
+	}
+
+	podUID := volCtx[podUIDContextKey]
+	if podUID == "" {
+		return nil, status.Error(codes.InvalidArgument, "No Pod UID included with request")
+	}
+
+	flavor := volCtx["flavor"]
+	if flavor == "" {
+		flavor = "default"
+	}
+	if flavor != "default" && flavor != "musl" {
```

Comment:
Maybe think about moving those flavours to consts since we reuse it a lot of times